pkg/scm/scm.go

```diff
@@ -103,15 +111,18 @@ func GenerateSCMToken(config *api.SCMConfig) error {
 				log.Errorf("fail to get SCM token for user %s as %s", config.Username, err.Error())
 				return err
 			}
+
+			// Update the token if generate a new one.
+			config.Token = token
+		} else if !provider.CheckToken(config) {
+			return fmt.Errorf("token is unauthorized to repos")
 		}
 	case api.SVN:
 		return fmt.Errorf("SCM %s is not supported", scmType)
 	default:
 		return fmt.Errorf("SCM type %s is unknow", scmType)
 	}
 
-	// Update the token if generate a new one.
-	config.Token = token
```

Comment:
Add a new variable `generatedToken` for generated token, and check its value here. If it is empty, then check the token of the config, otherwise set the generated token to the config. What about this implementation?